### PR TITLE
feat(social): enhance the social module

### LINKED ISSUE #559: request: Configurable channel for social achievements
**Is your feature request related to a problem? Please describe.**
When a user acquires a new autorole from the social monitor, Skyra has the option to send a message congratulating them, with the option to customise said message as well, but does not feature a setting to change its target channel.

**Describe the solution you'd like**
Add `social.achievement-channel` or `channels.achievements`.

**Describe alternatives you've considered**
None.

**Additional context**
None.

### PR DESCRIPTION
**Pull request by kyranet**
- feat(social): implement level notifications
- refactor(social): implement new tags syntax
- refactor(social): simplify achievement notification set-up
- refactor(social): split social awarding logic into different pieces

### TODO

- [x] `social.achieve-message` -> `social.achieve-role` (boolean to content)
- [x] `social.achieve-level` (boolean to content)
- [x] Remove default message (once the two previous steps are done).
- [x] Write migrations.
- [x] Transform all the messages to use the new tag syntax in the migrations.
- [x] Test extensively.
- [ ] ~~Update dashboard.~~ The page doesn't exist, so no update is needed.

### CODE REVIEW

File: services/Skyra.Database/Migrations/20210422211300_V03_NewSocialFeaturesAndValueRename.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace Skyra.Database.Migrations
+{
+    public partial class V03_NewSocialFeaturesAndValueRename : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.DropColumn(
+                name: "social.achieve",
```
Comment (vladfrangu) [inline services/Skyra.Database/Migrations/20210422211300_V03_NewSocialFeaturesAndValueRename.cs:10]: OOF. Any way to NOT drop this column? 😬

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: services/Skyra.Database/Migrations/20210422211300_V03_NewSocialFeaturesAndValueRename.cs (hunk 1 above)
Comment (kyranet) [reply services/Skyra.Database/Migrations/20210422211300_V03_NewSocialFeaturesAndValueRename.cs:10]: What do you mean?

**Review by Tylertron1998 — DISMISSED** [review]

File: services/Skyra.Database/Migrations/20210422211300_V03_NewSocialFeaturesAndValueRename.cs (hunk 1 above)
Comment (favna) [reply services/Skyra.Database/Migrations/20210422211300_V03_NewSocialFeaturesAndValueRename.cs:10]: He means that the data that is there right now is not being backed up but rather just completely pruned. This is not how we normally write migrations. Did you generate the migration _after_ putting some data in the database?

File: typescript/src/languages/en-US/settings.json [hunk 2]
```diff
@@ -118,8 +118,10 @@
 	"selfmodReactionsIgnoredRoles": "The roles that will be ignored by the reactions sub-system",
 	"selfmodReactionsMaximum": "The maximum amount of reactions before I will start applying penalties",
 	"selfmodReactionsAllowed": "The reactions that are allowed",
-	"socialAchieve": "Whether or not I should congratulate people who get a new leveled role.",
-	"socialAchieveMessage": "The congratulation message for people when they get a new leveled role. Requires `social.achieve` to be enabled.",
+	"socialAchieveRole": "Whether or not I should congratulate people who get a new leveled role.",
```
Comment (favna) [inline typescript/src/languages/en-US/settings.json]: Document all the tags similar to `birthday.message`

**Review by favna — CHANGES_REQUESTED** [review]

**Review by favna — DISMISSED** [review]

**Review by favna — APPROVED** [review]

**Pull request merged**